### PR TITLE
Test coverage

### PR DESCRIPTION
**Pull request by JasperTimm**
Fairly sizeable PR as it includes a whole bunch of new tests which improve test coverage quite a bit.

The following tests are added:
* Check role on authMint
* Check code already used in authMint
* Check valid use of sendGasOnAuth, also setSendGasOnAuth
* Check version()
* Check tokenVerificationURI, also set
* Check setBaseURI
* Check all 4 tokenId funs with non-existent tokens
* Check setRevoke and updateExpiry for role and non-existent token
* Check role on authorizeUpgrade, by upgrading from another acct
* Check supportsInterface, for both yes and no
* When sending eth the contract’s .call fails
    * Refunds
GSN
* Set trusted forwarder
    * Bad role
    * Actually sets forwarder, how to check?
* Check _msgData - buh bow (it's internal and we don't use it.... :()

Also a bunch of more advanced tests which uses TestWallets.sol when the receiver is a contract, such as re-entrancy e.t.c.

NB:- There are 2 small PRs which are outstanding, they should be merged to get an idea of what's only on this branch.

### CODE REVIEW

File: ethereum/kycdao-ntnft/test/KycdaoNTNFT.ts [hunk 1]
```diff
@@ -168,40 +273,183 @@ describe.only('KycdaoNtnft Membership', function () {
...
+    })
   })
 
-  describe('status', function () {
+  describe('getting metadata from KYCNFTs', function () {
+    describe('metadata', function () {
+      it('returns the expected tokenURI', async function () {
+        await memberNftAsMinter.authorizeMinting(123, anyone.address, testMetaUID, testVerifUID, expiration, false)
+        await memberNftAsAnyone.mint(123, {value: expectedMintCost})
+        expect(await memberNft.tokenURI(1)).to.equal(testInitArgs.baseURI + testMetaUID)
```
Comment (blackghost1987) [inline ethereum/kycdao-ntnft/test/KycdaoNTNFT.ts:391]: this should probably use memberNftAsAnyone to show that it's publicly available

File: ethereum/kycdao-ntnft/test/KycdaoNTNFT.ts [hunk 2]
```diff
@@ -168,40 +273,183 @@ describe.only('KycdaoNtnft Membership', function () {
...
+      it('fails when called with a non-existant tokenId', async function () {
+        await expect(memberNft.tokenURI(1)).to.be.revertedWith('ERC721Metadata: URI query for nonexistent token')
+      })
+    })
+
+    describe('verification metadata', function () {
+      it('returns the expected tokenVerificationURI', async function () {
+        await memberNftAsMinter.authorizeMinting(123, anyone.address, testMetaUID, testVerifUID, expiration, false)
+        await memberNftAsAnyone.mint(123, {value: expectedMintCost})
+        expect(await memberNft.tokenVerificationURI(1)).to.equal(testInitArgs.verificationBaseURI + testVerifUID)
```
Comment (blackghost1987) [inline ethereum/kycdao-ntnft/test/KycdaoNTNFT.ts:414]: this should use memberNftAsAnyone too

**Review by blackghost1987 — APPROVED** [review]
good work, just a minor suggestion

File: ethereum/kycdao-ntnft/test/KycdaoNTNFT.ts (hunk 1 above)
Comment (JasperTimm) [reply ethereum/kycdao-ntnft/test/KycdaoNTNFT.ts:391]: Yea good point! Will change them now. Thanks!

**Pull request merged**